### PR TITLE
istio: read network typology from cluster label

### LINKED ISSUE #47: support more complex network mode in istio
In https://github.com/kurator-dev/kurator/pull/46 we add non-flat mode about istio, but in a real world the typology of clusters may more complex.
thinking about a network typology like following:

istio-network-topology

we need a mechanism to support this but can be used in other plugin.

cc @hzxuzhonghu @prodanlabs

### PR DESCRIPTION
**Pull request by zirain**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #47 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
istio: read network typology from cluster label
```

### CODE REVIEW

File: pkg/plugin/istio/install.go [hunk 1]
```diff
@@ -486,8 +502,8 @@ func (p *IstioPlugin) iopFiles(cluster string) ([]string, error) {
 	iopFiles = append(iopFiles, p.args.IopFiles...)
 	if !p.IsFlat() {
 		mesh := clusterNetwork{
-			Network:     networkName(cluster), // TODO: support custom network, e.g. primary and remote1 in network1, remote2 in network2
-			MeshID:      "mesh1",              // TODO: make this configurable
+			Network:     p.clusterNetwork(cluster),
```
Comment (hzxuzhonghu) [inline pkg/plugin/istio/install.go:505]: why remove the TODO，IMO this is not resoloved by this patch

File: pkg/plugin/istio/install.go [hunk 2]
```diff
@@ -477,7 +478,22 @@ func (p *IstioPlugin) installRemotes(remotePilotAddress string) error {
 	return multiErr.ErrorOrNil()
 }
 
+func (p *IstioPlugin) clusterNetwork(cluster string) string {
+	c, err := p.KarmadaClient().ClusterV1alpha1().Clusters().Get(context.TODO(), cluster, metav1.GetOptions{})
+	if err != nil {
+		// shhould not happen, fallback
```
Comment (hzxuzhonghu) [inline pkg/plugin/istio/install.go:484]: should

Comment (hzxuzhonghu) [reply pkg/plugin/istio/install.go:484]: This can happen

File: pkg/plugin/istio/install.go (hunk 1 above)
Comment (zirain) [reply pkg/plugin/istio/install.go:505]: mistake

Comment (zirain) [reply pkg/plugin/istio/install.go:505]: you can label on cluster before run `kurator install istio` to achieve this

Comment (hzxuzhonghu) [reply pkg/plugin/istio/install.go:505]: I mean do you support 

`e.g. primary and remote1 in network1, remote2 in network2`

From my reading, it seems even `primary and remote1 in network1`, we still added east-west gateway

Comment (zirain) [reply pkg/plugin/istio/install.go:505]: yes, it should be improved, let's track that with an issue.

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**